### ISSUE TITLE
Bug in EtherValidator implementation of = (span equality op)

### ISSUE THREAD

**Issue by amir-zeldes** [gucorpling/gitdox#107]
I'm seeing the validator miss unequal spans, I think the problem is that in this part:

https://github.com/gucorpling/gitdox/blob/dev/modules/validation/ether_validator.py#L97-L114

The rule object spreads the value of a span into all constituent cells, then does the name/arg version of:

`name_tuples[row].append((letter, cell.content))`

But since row changes each time, none of these tuple rows get a list longer than 1 element. Then to check the span length in _apply_equal_span_length (or same with content), it does:

`name_len = len(name_tuples[row])`
https://github.com/gucorpling/gitdox/blob/dev/modules/validation/ether_validator.py#L160

But I think this ends up being 1 regardless of length.

**Comment by lgessler** [comment]
Will take a look at this soon--do you have an example that doesn't validate properly handy?

**Comment by amir-zeldes** [comment]
Yes, try a22.YB307-320, for the rule translation=verse. There are some definite violations in the document, which are causing export validation to fail, but are not caught in the spreadsheet. Thanks!

**Comment by lgessler** [comment]
I wasn't able to successfully import that text into my dev machine, but I think I found the problem. If I saw the right thing, the problem wasn't actually with the validation itself, but rather the highlighting mechanism. 

To see if this is indeed your same problem, you could check to see whether the problematic cells are getting noted in the output. I think they should be getting noted in the output, but that their cells should not be getting highlighted. My patch fixes this (#106).

Let me know if the problem still isn't fixed.

**Comment by lgessler** [comment]
@amir-zeldes, I put that document into my dev instance but the rule translation = verse appears to be satisfied. Perhaps when you get a moment you could send me a paste of what validation errors for this rule you're seeing?

Also, another thing I noticed: in the version of this doc that's on scriptorium, row 2 is entirely empty (even the token slot). This causes a function in ether.py to choke, causing validation to fail.

**Comment by lgessler** [comment]
Nevermind, there was indeed a bug, at least with this case where two cols have the same single-cell coverage but different cell spans:

```
_ _
A C
A C
B C
B D
B D
_ _
```

Should be fixed now.

**Comment by amir-zeldes** [comment]
Yes, the latest PR fixes this - thanks! The conflicting spans are being caught now. If it's very inefficient then refactoring is important, since validating 'everything' (which needs to happen at least when the export profile is edited) does take very long. But if it's mainly aesthetics without a substantial performance hit, holding off on this should be fine. In terms of quality of life improvements, non-cgi asynchronous metadata editing might be the highest value.
